src/main/java/org/apache/maven/plugins/gpg/GpgVersion.java

```diff
@@ -140,4 +140,26 @@ public String toString()
         return rawVersion;
     }
 
+    @Override
+    public boolean equals( Object o )
+    {
+        if ( this == o )
+        {
+            return true;
+        }
+
+        if ( o == null || getClass() != o.getClass() )
+        {
+            return false;
+        }
+
+        GpgVersion that = (GpgVersion) o;
+        return compareTo( that ) == 0;
```

Comment:
as written compareTo can throw an IllegalArgumentException which violates the contract of equals. I'm not sure if you can just compare the raw strings here or if you need to parse the string in the constructor first.

Reply:
Forced pushed the try catch for IllegalArgumentException.
But this `compareTo` does not compare a raw string. I think the equals should not necessarily compare strings, but the what is represented byt the object - GPG version.